elastic/apm-server
# aggregation/txmetrics: drop user-agent parsing

## Motivation/summary

Remove user-agent from transaction metrics, and all related config and processing. RUM-specific charts have been removed from the APM app, and these metrics are not currently required by the User Experience app. Drop the TODOs about adding country name while we're at it.

## Checklist

~- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~ (not yet user-facing)

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

1. Run server with transaction metrics enabled.
2. Send some RUM page-loads: same page with different browsers.
3. Observe that transaction metrics are aggregated, but user-agent is not included and the page-loads from multiple browsers are grouped together.

## Related issues

https://github.com/elastic/apm-server/issues/3613#issuecomment-727681970
https://github.com/elastic/kibana/pull/77458
